rfcs/mson-type-inference.md

```diff
@@ -0,0 +1,77 @@
+---
+RFC: XXXX
+Author: Kyle Fuller
+Status: Draft
+Created: 2016-01-15
+Last Modified: 2016-02-28
+---
+
+# MSON RFC XXXX: Type Inference
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Rationale](#rationale)
+- [Backwards Compatibility](#backwards-compatibility)
+
+## Abstract
+
+This proposal adds type inference to MSON, so that MSON parsers can determine
+an appropriate type to use for each MSON member instead of defaulting to
+`string`.
+
+In short, the following example would result in a property called
+`writable` which is of the type `boolean` and has a sample value of `true`.
+
+```apib
++ writable: true
+```
+
+Previously, the above example would have resulted in a `string` with the
+sample value `true`.
+
+## Motivation
+
+Type inference would simplify many uses of MSON and allow users to be less
+explicit about the types. This makes the language more intuitive and simpler
+for new users.
+
+## Rationale
+
+Below outlines the types of values that would be type-inferred.
```

Comment:
Should we include a `null` inferred type? That would cover all the JSON types then.